### PR TITLE
Fix DynamicTable issues

### PR DESCRIPTION
**Pull request by ajtritt**
## Motivation

1. Fix constructing DynamicTable with columns as DataChunkIterators
2. Add ability to disable converting DynamicTable selections into Pandas DataFrame

### CODE REVIEW

**Review by oruebel — DISMISSED** [review]

File: src/hdmf/common/table.py [hunk 1]
```diff
@@ -46,6 +46,9 @@ def add_row(self, **kwargs):
         val = getargs('val', kwargs)
         self.append(val)
 
+    def get(self, key, **kwargs):
+        return super().get(key)
+
```
Comment (rly) [inline src/hdmf/common/table.py:51]: Is this function necessary?

Comment (rly) [reply src/hdmf/common/table.py:51]: Oh, I see. Never mind

File: src/hdmf/common/table.py [hunk 2]
```diff
@@ -122,27 +125,22 @@ def __getitem_helper(self, arg, **kwargs):
...
         end = self.data[arg]
         return self.target.get(slice(start, end), **kwargs)
 
     def __getitem__(self, arg):
-        """
-        Select elements in this VectorIndex and retrieve the corrsponding data from the self.target VectorData
-
-        :param arg: slice or integer index indicating the elements we want to select in this VectorIndex
-        :return: Scalar or list of values retrieved
-        """
```
Comment (rly) [inline src/hdmf/common/table.py]: Why remove the docstring? This function is still included in the API documentation.

**Review by rly — DISMISSED** [review]

**Comment by rly** [comment]
Please update `CHANGELOG.md` describing the changes. I think this would fall into a bugfix release 2.1.1.

**Review by rly — APPROVED** [review]

**Pull request merged**